### PR TITLE
Refactor snyk component

### PR DESCRIPTION
**Pull request by mikomraz**

### CODE REVIEW

File: lib/java_buildpack/framework/snyk.rb [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+      def find_main_pom
+        Dir.glob("#{@droplet.root}/**/pom.xml")[0]
+      end
+
+      def all_jars
+        Dir.glob("#{@droplet.root}/WEB-INF/**/*.jar")
+      end
+
+      def poms_from_jar(jar)
+        ret = []
```
Comment (AH7) [inline lib/java_buildpack/framework/snyk.rb]: Leftover

File: lib/java_buildpack/framework/snyk.rb [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+        issues = sort_issues(issues)
+        if severity_threshold
+          issues = filter_issues(issues, severity_threshold)
+        end
+        
+        issues.each do |issue|
+          print_issue(issue)
+        end
+
+        # puts "#{'Organisation:'.bold}      #{test['org']['name']}"
```
Comment (AH7) [inline lib/java_buildpack/framework/snyk.rb]: remove comments

File: lib/java_buildpack/framework/snyk.rb [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+        # puts "#{'Target file:'.bold}       -"
+        # puts "#{'Open source:'.bold}       ?"
+        # puts "#{'Project path:'.bold}      ?"
+
+        dependencyCount = test['dependencyCount'] || 0
+        if issues.length == 0
+          @logger.info "✓ Tested #{dependencyCount} dependencies for known vulnerabilities, no vulnerable paths found\n".green
+        else
+          unique_count = issues.map{ |issue| issue['id']}.uniq.length
+          vulnerable_paths = issues.map{ |issue| issue['from']}.flatten.uniq.length
```
Comment (AH7) [inline lib/java_buildpack/framework/snyk.rb]: Is that accurate? what happens if we have 2 diff vulns on the same indirect deps path?

File: lib/java_buildpack/framework/snyk.rb [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+          return "", []
+        end
+      end
+
+      def find_main_pom
+        Dir.glob("#{@droplet.root}/**/pom.xml")[0]
+      end
+
+      def all_jars
+        Dir.glob("#{@droplet.root}/WEB-INF/**/*.jar")
```
Comment (AH7) [inline lib/java_buildpack/framework/snyk.rb]: How a gradle droplet looks like? Does it also contain the *.jar under WEB-INF?

File: lib/java_buildpack/framework/snyk.rb [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+
+        dependencyCount = test['dependencyCount'] || 0
+        if issues.length == 0
+          @logger.info "✓ Tested #{dependencyCount} dependencies for known vulnerabilities, no vulnerable paths found\n".green
+        else
+          unique_count = issues.map{ |issue| issue['id']}.uniq.length
+          vulnerable_paths = issues.map{ |issue| issue['from']}.flatten.uniq.length
+          @logger.info "\e[0mTested #{dependencyCount} dependencies for known vulnerabilities, #{"found #{unique_count} vulnerabilities, #{vulnerable_paths} vulnerable paths.".red}\n"
+        end
+        @logger.info " "
```
Comment (AH7) [inline lib/java_buildpack/framework/snyk.rb]: Why do we need this? \n is not enough?

File: lib/java_buildpack/framework/snyk.rb [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+      def severity_score(severity)
+        return {
+          'high' => 3,
+          'medium' => 2,
+          'low' => 1
+        }[severity]
+      end
+
+      def sort_issues(issues)
+          severityMap = {
```
Comment (AH7) [inline lib/java_buildpack/framework/snyk.rb]: Leftover?

Comment (mikomraz) [reply lib/java_buildpack/framework/snyk.rb]: yes

File: lib/java_buildpack/framework/snyk.rb (hunk 5 above)
Comment (mikomraz) [reply lib/java_buildpack/framework/snyk.rb]: I changed everything to " ". Using "\n" scrambled the logger output

File: lib/java_buildpack/framework/snyk.rb (hunk 3 above)
Comment (mikomraz) [reply lib/java_buildpack/framework/snyk.rb]: ok. is it correct to do uniq on the vuln id?

Comment (mikomraz) [reply lib/java_buildpack/framework/snyk.rb]: ok, I removed uniq. Is it still correct for vuln ids?

File: lib/java_buildpack/framework/snyk.rb (hunk 5 above)
Comment (mikomraz) [reply lib/java_buildpack/framework/snyk.rb]: \n scarmabled logger output. removed all \n

**Pull request merged**